src/responder/nss/nss_enum.c

```diff
@@ -138,7 +138,7 @@ static void nss_setent_internal_done(struct tevent_req *subreq)
     switch (ret) {
     case EOK:
         talloc_zfree(state->enum_ctx->result);
-        state->enum_ctx->result = talloc_steal(state->nss_ctx, result);
+        state->enum_ctx->result = talloc_steal(state->enum_ctx, result);
```

Comment:
I understand how is this better for netgroups where the enum_ctx is a child of the hash entry keyed by the netgroup name, but I'm struggling to see how is this handled with regular enumeration..
I see some talloc_free() calls in nss_setent_internal_done() but those were there before. Do you know where are the user or group enumeration results freed? Or how come we only see the issue with netgroups?

Reply:
For netgroup, we create one `enum_ctx` for each requested netgroup. This whole object is freed when timeout occurs (nss_setnetgrent_timeout`).

For passwd, groups and other, the `enum_ctx` stays for lifetime of the responser and only `enum_ctx->result` is freed when timeout occurs (`nss_setent_timeout`).

For all, `enum_ctx->result` is freed when new result is available in `nss_setent_internal_done`.